### PR TITLE
Add support for validation of dates

### PR DESCRIPTION
**Pull request by petedmarsh**
Swagger defines the date format, which are date (no time) strings
conforming to the full date format defined in RFC3339 (essentially
just the date portion of a full ISO8601 date/time). This adds
support for validation of this format.

### CODE REVIEW

File: flex/error_messages.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 FORMAT_MESSAGES = {
     'invalid': "Value {0} does not conform to the format {1}",
     'invalid_uuid': "{0} is not a valid uuid",
+    'invalid_date': "{0} is not a valid RFC3339 full-date",
```
Comment (petedmarsh) [inline flex/error_messages.py]: Perhaps this should say IS8601 date, - RFC339 is what Swagger says but inside there it's refering to ISO8601 in any case

Comment (pipermerriam) [inline flex/error_messages.py]: I'd suggest putting both `RFC3339/ISO8601`

**Comment by petedmarsh** [comment]
Shall I make a similar chance to the date time message too?

**Comment by pipermerriam** [comment]
> Shall I make a similar chance to the date time message too?

If you don't mind :)

**Comment by petedmarsh** [comment]
I've updated the error messages

**Comment by pipermerriam** [comment]
You can get this in `6.7.0`

**Pull request merged**